server/py/services/api/api/endpoints/alert_activations.py

```
@@ -40,7 +40,7 @@ async def list_alert_activations(
     entity: Optional[str] = None,
     severity: Optional[
         list[Union[mlrun.common.schemas.alert.AlertSeverity, str]]
```

Actually, why do we support a random string at all?
The idea of using the schemas is that we will have validations on the input and of won't be able to send something random.

---

@TomerShor firstly, This is just typing.. from real-world side it can be anything. Also, it is for filtering, so I thought that allowing filtering by string is better for any BC issues, but if you think we need it, I can explicitly translate it to AlertSeverity object. However I haven't found any place in the code where we would do it like this.

---

In FastAPI, pydantic is enforcing the endpoint params types, so typing does matter.
There is not BC here as this is a new endpoint, and the severity is also enforced on the [AlertConfig object](https://github.com/mlrun/mlrun/blob/1b288a97709161603f2586ada547ac1db346ede4/mlrun/common/schemas/alert.py#L151), so you can't set it to a random string.
Lastly, I think it is better to fail-fast if user tries to filter on something that we don't support

---

@TomerShor we don't translate the given object to AlertConfig/AlertSeverity when listing alert activations. In terms of BC I was talking about another case. Imagine client has newer version of MLRun which support's new type of AlertSeverity. If user passes this new type in the filter along with other severities, we will fail fast. Do we want to do this instead of just ignoring it?

---

If the client is newer then the server, this new severity won't be supported in the backend anyway.
IMO if we choose using the schemas as the query param types, then we should enforce them. If we don't want to enforce specific values then why use schemas at all?..

---

@TomerShor I agree that we should have validation when storing data. But when filtering.. Especially if we allow the list here, I'd go with best-effort approach

---

Also, it is out of scope for this PR. Whether we decide to hardcode schemas here or not, it can be done in separate PR.